test/integration/util/create-cluster.sh

```diff
@@ -13,6 +13,36 @@ containerdConfigPatches:
 - |-
   [plugins.\"io.containerd.grpc.v1.cri\".registry.mirrors.\"$REGISTRY_NAME:$REGISTRY_PORT\"]
     endpoint = [\"http://${REGISTRY_NAME}:${REGISTRY_PORT}\"]
+kubeadmConfigPatches:
+- |
+  apiVersion: kubeadm.k8s.io/v1beta2
+  kind: ClusterConfiguration
+  metadata:
+    name: config
+  apiServer:
+    extraArgs:
+      "feature-gates": "IPv6DualStack=true"
+      "service-cluster-ip-range": "10.96.0.0/16,fd00::/108"
+  controllerManager:
+    extraArgs:
+      "feature-gates": "IPv6DualStack=true"
+      "service-cluster-ip-range": "10.96.0.0/16,fd00::/108"
+      "cluster-cidr": "10.244.0.0/16,fc00::/48"
+      "node-cidr-mask-size": \"0\"
```

Comment:
When testing this configuration locally, I received:

```
Error: invalid argument "\\\"0\\\"" for "--node-cidr-mask-size" flag: strconv.ParseInt: parsing "\\\"0\\\"": invalid syntax
```

From the `kube-controller-manager`, which consequently was stuck in crash loop due to this.

I believe the [defaults](https://kubernetes.io/docs/concepts/services-networking/dual-stack/#enable-ipv4-ipv6-dual-stack) in this case should be sufficient, but I noticed when testing this that kind appears to use the flag by default which is incompatible with dual stack:

```
controllermanager.go:249] error starting controllers: usage of --node-cidr-mask-size is not allowed with dual-stack clusters
```

This is something that needs to be improved upstream in [kind](https://github.com/kubernetes-sigs/kind) but I take it you were trying to override it to work around this for now.

I'm looking further into this and will report back.

Reply:
So I ended up tracking this down through `kind` and into the `kubeadm` codebase:

In the [Kubeadm ClusterConfiguration Specification](https://pkg.go.dev/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm#ClusterConfiguration) I found that you must set a topical `featureGates` option, so that your kind config ends up looking something like this:

```yaml
apiVersion: kind.x-k8s.io/v1alpha4
kind: Cluster
name: testing-prs
kubeadmConfigPatches:
- |
  apiVersion: kubeadm.k8s.io/v1beta2
  kind: ClusterConfiguration
  featureGates:
    IPv6DualStack: true
  metadata:
    name: config
```

With the above in my configuration I was able to get the cluster to deploy successfully, and by the looks of the code if you set this topically on the `ClusterConfiguration` it will automatically configure it for Kubernetes components.

Test that out and let me know what you find :vulcan_salute: